syslog/syslog.go

```diff
@@ -90,25 +90,20 @@ type Logger struct {
 
 // Dial connects to the syslog server at raddr, using the optional certBundle,
 // and launches a goroutine to watch logger.Packets for messages to log.
-func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger, error) {
-	// dial once, just to make sure the network is working
-	conn, err := dial(network, raddr, rootCAs)
-
-	if err != nil {
-		return nil, err
-	} else {
-		logger := &Logger{
-			ClientHostname: clientHostname,
-			network:        network,
-			raddr:          raddr,
-			rootCAs:        rootCAs,
-			Packets:        make(chan Packet, 100),
-			Errors:         make(chan error, 0),
-			conn:           conn,
-		}
-		go logger.writeLoop()
-		return logger, nil
+func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger) {
+	logger := &Logger{
+		ClientHostname: clientHostname,
+		network:        network,
+		raddr:          raddr,
+		rootCAs:        rootCAs,
+		Packets:        make(chan Packet, 100),
+		Errors:         make(chan error, 0),
 	}
+
+	logger.connect()
```

Comment:
Why do we care about retrying forever if there isn't anything to send? What is the benefit of ensuring that a connection is always established if there isn't something that needs to be sent?